runtime/near-vm-runner/src/tests/runtime_errors.rs

```diff
@@ -873,3 +883,297 @@ fn test_nan_sign() {
         }
     });
 }
+
+// Check that a `GasExceeded` error is returned when there is not enough gas to
+// even load a contract.
+#[test]
+fn test_gas_exceed_loading() {
+    with_vm_variants(|vm_kind: VMKind| {
+        // Loading is the first thing that happens on the VM entrypoint that
+        // requires gas, thus attaching only 1 gas triggers the desired
+        // behavior.
+        let gas = 1;
+        let method_name = "non_empty_non_existing";
+        let actual =
+            make_simple_contract_call_with_gas_vm(&simple_contract(), method_name, gas, vm_kind);
+        gas_and_error_match(
+            actual,
+            Some(gas), //< It is import to be non-zero gas here.
+            Some(VMError::FunctionCallError(FunctionCallError::HostError(HostError::GasExceeded))),
+        );
+    });
+}
+
+#[cfg(feature = "protocol_feature_fix_contract_loading_cost")]
+mod fix_contract_loading_cost_protocol_upgrade {
+    use super::*;
+    use crate::tests::make_simple_contract_call_ex;
+
+    /// Simple contract that uses non-zero gas to execute main.
+    #[cfg(feature = "protocol_feature_fix_contract_loading_cost")]
+    fn almost_trivial_contract_and_exec_gas() -> (Vec<u8>, u64) {
+        let code = wat::parse_str(
+            r#"
+                        (module
+                          (type (;0;) (func))
+                          (func (;0;) (type 0)
+                            i32.const 1
+                            i32.const 1
+                            i32.div_s
+                            return
+                          )
+                          (export "main" (func 0))
+                        )"#,
```

Comment:
For new code, let's try to stick with human-readable wat, without cryptic `(;0;)`:

```
(module
  (func (export "main")
    i32.const 1
    i32.const 1
    i32.div_s
    return
  )
)
```

Reply:
Done